components/apimgt/org.wso2.carbon.apimgt.keymgt/src/test/java/org/wso2/carbon/apimgt/keymgt/handlers/DefaultKeyValidationHandlerTest.java

```diff
@@ -0,0 +1,148 @@
+/*
+ *
+ *   Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *   WSO2 Inc. licenses this file to you under the Apache License,
+ *   Version 2.0 (the "License"); you may not use this file except
+ *   in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
+package org.wso2.carbon.apimgt.keymgt.handlers;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.wso2.carbon.apimgt.api.model.subscription.URLMapping;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+import org.wso2.carbon.apimgt.impl.dto.APIKeyValidationInfoDTO;
+import org.wso2.carbon.apimgt.keymgt.APIKeyMgtException;
+import org.wso2.carbon.apimgt.keymgt.SubscriptionDataHolder;
+import org.wso2.carbon.apimgt.keymgt.model.SubscriptionDataStore;
+import org.wso2.carbon.apimgt.keymgt.model.entity.API;
+import org.wso2.carbon.apimgt.keymgt.service.TokenValidationContext;
+import org.wso2.carbon.base.MultitenantConstants;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static org.mockito.Matchers.eq;
+import static org.wso2.carbon.base.CarbonBaseConstants.CARBON_HOME;
+
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({DefaultKeyValidationHandler.class, TokenValidationContext.class, APIKeyValidationInfoDTO.class,
+        SubscriptionDataStore.class, SubscriptionDataHolder.class, PrivilegedCarbonContext.class})
+
+public class DefaultKeyValidationHandlerTest extends DefaultKeyValidationHandler {
+
+    private final String USER_NAME = "admin";
+    private final String API_CONTEXT = "/apicontext/1.0.0";
+    private final String API_NAME = "test_api";
+    private final String DEFAULT_API_NAME = "default_api";
```

Comment:
I think we don't need another API Name for the default API because this is the same API we are testing the 2 scenarios against. Let's use API_NAME instead.